src/TestingUtils/Microsoft.AspNetCore.Testing/test/FlakyAttributeTest.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.AspNetCore.Testing.xunit;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Testing.Tests
+{
+    public class FlakyAttributeTest
+    {
+        [Fact]
+        [Flaky("http://example.com")]
+        public void AlwaysFlaky()
+        {
+            throw new Exception("Flakey!");
+        }
+
+        [Fact]
+        [Flaky("http://example.com", OnAzDO = false)]
+        public void FlakyInHelixOnly()
+        {
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("HELIX")))
+            {
+                throw new Exception("Flaky on Helix!");
+            }
+        }
+
+        [Fact]
+        [Flaky("http://example.com", OnAzDO = false, OnHelixQueues = HelixQueues.macOS1012Amd64 + HelixQueues.Fedora28Amd64)]
+        public void FlakyInSpecificHelixQueue()
+        {
+            var queueName = Environment.GetEnvironmentVariable("HELIX");
```

Comment:
@HaoK I noticed aspnet/AspNetCore#8231 is adding this env var, can you double-check that my logic here seems sound? The idea is to actually make this test properly fail on the specified Helix queues so we can leave it in as validation that flaky tests are indeed treated properly, so I want this test to actually be correctly detecting the queue it's running in :)

Reply:
Might be time to have a HelixUtils to consolidate all the helix stuff in one central file, or you could use SkipOnHelixAttribute.GetTargetHelixQueue() when that PR is merged.

Logic looks fine. One other pivot we should probably consider, is adding the concept of Platform as well, so there's an easy way to mark `Flaky(OnlyPlatforms = "OSX;Linux")` so we don't have to enumerate every queue

So maybe we need:
```
   public class HelixQueue {
       public string Id;
       public string Platform
   }
   public static class HelixPlatforms {
       public string Windows = "Windows";
       public string OSX = "OSX";
       public string Linux = "Linux";
   }
```

Reply:
We could just do that by building combo-`const`s like so:

```csharp
        // Common Groupings
        public const string Ubuntu = Ubuntu1604Amd64 + Ubuntu1810Amd64;
        public const string Centos = Centos7Amd64;
        public const string Debian = Debian8Amd64 + Debian9Amd64;
        public const string Redhat = Redhat7Amd64;
        public const string Fedora = Fedora27Amd64 + Fedora28Amd64;

        public const string Windows = Windows10Amd64;
        public const string macOS = macOS1012Amd64;
        public const string Linux = Ubuntu + Centos + Debian + Redhat + Fedora;
```

We could also add other pivots to the flaky traits. I already changed it a bit so that the trait we insert is `Flaky:Helix:Queue:[queuename]`, we could have other facets in that sequence and as long as the script running the tests injects all the necessary filters, it should work.